### DISCUSSION TITLE
useFetch not working during pre-render

### DISCUSSION THREAD

**Discussion by AlexMili** [nuxt/nuxt#28448] [Questions] [answered]
I am using `useFetch` with `server: true` to get data from a server's API.
When I run the command `yarn nuxt generate --prerender` I see on the server's logs that the pre-render process do send a request to the server. However, when I see the generated result, I still see my request being sent to the server without using the pre-rendered data.

I have two questions:
- How is it possible that the client is doing the request since I set `server` to `true`?
- Why is my build is not using the pre-rendered data?

Here is my function:
```ts
export const requestAPI = async function(url: string) {
    return await new Promise<any>((resolve: any, reject: any) => {
        useFetch(url, {
            server: true,
            onResponseError({ error }) { reject(error); },
            onResponse({ response }) {
                const { status, _data } = response;
                resolve(_data);
            }
        });
    });
};
```

And my vue:
```js
<script setup lang="ts">
    const items = ref([]);

    requestAPI("http://127.0.0.1/api/items").then(data => {
        items.value = data;
    });

</script>
<template>
<div v-for="item in items" :key="item.id">
{{ item.id }}
</div>
</template>
```

**Comment by xjccc** [comment]
useFetch return a promise

```
export const requestAPI = async function(url: string) {
    return  useFetch(url, {
            // default true
            server: true
    })
}
```

```
<script setup lang="ts">
    const items = ref([])
    const {data} = await requestAPI("http://127.0.0.1/api/items")
     items.value = data
</script>
```

**Reply by AlexMili** [reply]
Thanks I tested on a new dummy project to avoid all the complexity of my project and `useFetch` with your code technically works but I don't have anything rendered. I think I am missing something. Here is my code:
```javascript
<script lang="ts" setup>
    const items = ref<any>([])
    const { data } = await useFetch("http://127.0.0.1/api/items", {
            // default true
            server: true,
    })
    items.value = data;

</script>
<template>
    <div>
        <div v-for="item in items" :key="item?.id">{{ item?.id}}</div>
    </div>
</template>
```

If I don't put `item?.id` I get an error saying that `id` doesn't exist on undefined object. And with the above code, nothing is rendered, I don't have anything in my HTML. But this time I don't have the request made by the client so this is a step forward!

**Reply by xjccc** [reply]
I think you can use [stackblitz](https://stackblitz.com/) for a small repo.

**Comment by AlexMili** [comment, marked as answer]
Finally found the issue. I am using Vuetify and for whatever reason SSR is not working when `NuxtPage` is encapsulated in `VMain` directive. I just removed the `VMain` tag of my `app.vue` and everything is working smoothly.
I opened an issue on Vuetify's side: https://github.com/vuetifyjs/vuetify/issues/20300
